### ISSUE TITLE
Label Smoothing with Binary objective (log loss)

### ISSUE THREAD

**Issue by carlosg-m** [microsoft/LightGBM#6716]
## Description
LightGBM Binary objective does not seem to support soft labels. The point is to implement Label Smoothing.

For example:
label_smoothing = 0.2
y = (1 - label_smoothing) * y + label_smoothing / number_classes

| y |  y_smoothed |
|---|---------------|
0  |  0.1
1  |  0.9

**Comment by carlosg-m** [comment]
Found out by myself. Had to use `cross_entropy` as the objective.
```
cross-entropy application

- cross_entropy, objective function for cross-entropy (with optional linear weights), aliases: xentropy

- cross_entropy_lambda, alternative parameterization of cross-entropy, aliases: xentlambda

- label is anything in interval [0, 1]
```
